Makefile

```diff
@@ -0,0 +1,45 @@
+# Disable make's implicit rules, which are not useful for golang, and
+# slow down the build considerably.
+
+VPATH=bin:plugins/l4lb:pkg/spartan
+
+# Default go OS to linux
+export GOOS?=linux
+
+# Set $GOPATH to a local directory so that we are not influenced by
+# the hierarchical structure of an existing $GOPATH directory.
+export GOPATH=$(shell pwd)/gopath
+
+#dcos-l4lb
+L4LB=github.com/dcos/dcos-cni/plugins/l4lb
+L4LB_SRC=$(wildcard plugins/l4lb/*.go) $(wildcard pkg/spartan/*.go)
+
+PLUGINS=dcos-l4lb
+
+.PHONY: all plugin
+default: all
+
+.PHONY: clean
+clean:
+	rm -rf vendor bin gopath
+
+gopath:
+	mkdir -p gopath/src/github.com/dcos
+	ln -s `pwd` gopath/src/github.com/dcos/dcos-cni
```

Comment:
This would work only with if you are in the source directory. Would't work with `cwd` of the Makefile, will see I can make it more generic.